File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncgroupsdialog/SyncGroupsDialogMvpView.java
package com.mifos.mifosxdroid.dialogfragments.syncgroupsdialog;

import com.mifos.mifosxdroid.base.MvpView;

/**
 * Created by Rajan Maurya on 11/09/16.
 */
public interface SyncGroupsDialogMvpView extends MvpView {

    void showUI();

    void showSyncingGroup(String clientName);

    void showSyncedFailedGroups(int failedCount);

    void setMaxSingleSyncGroupProgressBar(int total);

    void setClientSyncProgressBarMax(int count);

    void updateClientSyncProgressBar(int i);

    void updateSingleSyncGroupProgressBar(int i);

    void updateTotalSyncGroupProgressBarAndCount(int i);

    int getMaxSingleSyncGroupProgressBar();

    void showNetworkIsNotAvailable();

    void showGroupsSyncSuccessfully();

    Boolean isOnline();

    void dismissDialog();

    void showDialog();

    void hideDialog();

    void showError(int s);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncgroupsdialog/SyncGroupsDialogMvpView.kt
package com.mifos.mifosxdroid.dialogfragments.syncgroupsdialog

import com.mifos.mifosxdroid.base.MvpView

/**
 * Created by Rajan Maurya on 11/09/16.
 */
interface SyncGroupsDialogMvpView : MvpView {
    fun showUI()
    fun showSyncingGroup(clientName: String?)
    fun showSyncedFailedGroups(failedCount: Int)
    fun setClientSyncProgressBarMax(count: Int)
    fun updateClientSyncProgressBar(i: Int)
    fun updateSingleSyncGroupProgressBar(i: Int)
    fun updateTotalSyncGroupProgressBarAndCount(i: Int)
    var maxSingleSyncGroupProgressBar: Int
    fun showNetworkIsNotAvailable()
    fun showGroupsSyncSuccessfully()
    val isOnline: Boolean?
    fun dismissDialog()
    fun showDialog()
    fun hideDialog()
    fun showError(s: Int)
}